### PR TITLE
Tidy

### PR DESCRIPTION
**Pull request by vext01**
Small tidy ups and improvements to the warmup experiment. See individual commit messages for details.

OK?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,7 +4,9 @@ work
 **/*.swp
 **/*.pyc
 **/*.class
+**/*.so
 warmup_results.json
 benchmarks/richards/java/Benchmark.java
 benchmarks/richards/java/Program.java
 benchmarks/richards/java/richards.java
+benchmarks/richards/java/richards.java.orig
```
Comment (ltratt) [inline .gitignore:12]: I don't think we should be excluding `.orig` files, because we don't ever generate them, right?

**Comment by ltratt** [comment]
Just one minor comment, then we can merge.

File: .gitignore (hunk 1 above)
Comment (vext01) [inline .gitignore:12]: `build.sh` creates this file when applying `java_richards.patch`, so we do create this file.

Comment (ltratt) [inline .gitignore:12]: Ah, this is due to a BSD `patch` oddity :( OK...

**Pull request merged**